packages/@lwc/features/src/babel-plugin/index.js

```diff
@@ -97,18 +92,27 @@ module.exports = function({ types: t }) {
                 }
 
                 const name = propertyPath.node.name;
-                let value = this.featureFlags[name];
+                let value = (this.opts.featureFlags || defaultFeatureFlags)[name];
                 validate(name, value);
 
-                if (!this.opts.prod || value === null) {
+                if (!this.opts.prod) {
                     if (isCompileTimeFlag) {
                         testPath.node.object = t.identifier(RUNTIME_FLAGS_IDENTIFIER);
-                        return;
                     }
+                    return;
                 }
 
                 if (this.opts.prod) {
-                    if (isUnaryNegation) {
+                    // This transform is generally run in prod-mode, after
+                    // running it beforehand in non-prod mode. This logic
+                    // handles the case where the code was not previously
+                    // transformed in non-prod mode.
+                    if (isCompileTimeFlag && value === null) {
+                        testPath.node.object = t.identifier(RUNTIME_FLAGS_IDENTIFIER);
+                        return;
+                    }
+
+                    if (isUnaryNegation && typeof value === 'boolean') {
```

Comment:
This is the bug fix. Validate that the value is a boolean before processing it as a boolean. This was causing `null` values to be processed as `false`.